Misc/NEWS.d/next/Core and Builtins/2024-05-12-03-10-36.gh-issue-118950.5Wc4vp.rst

```diff
@@ -0,0 +1,3 @@
+Let _SSLProtocolTransport.is_closing reflect SSLProtocol internal transport
+closing state so that StreamWriter.drain will invoke sleep(0) which calls
+connection_lost and correctly notifies waiters of connection lost.
```

Comment:
Fix bug where SSLProtocol.connection_lost wasn't getting called when OSError was thrown on writing to socket.